CleanRepo/Program.cs

```diff
@@ -499,19 +499,41 @@ private static void ListOrphanedTopics(List<FileInfo> tocFiles, List<FileInfo> m
 
                 if (!found)
                 {
-                    countNotFound++;
-                    output.AppendLine(markdownFile.FullName);
-
-                    // Delete the file if the option is set.
+                    // Try to delete the file if the option is set.
                     if (deleteOrphanedTopics)
                     {
-                        output.AppendLine($"DELETING '{markdownFile.FullName}'.");
-                        File.Delete(markdownFile.FullName);
+                        var isLinked = false;
+                        var referencedFile = "";
+                        foreach (var otherMarkdownFile in markdownFiles.Where(file => file != markdownFile))
+                        {
+                            if (!IsFileLinkedInFile(markdownFile, otherMarkdownFile))
+                            {
+                                continue;
+                            }
+
+                            referencedFile = otherMarkdownFile.FullName;
+                            isLinked = true;
+                            break;
+                        }
+
+                        if (isLinked)
+                        {
+                            output.AppendLine($"Unable to delete '{markdownFile.FullName}'");
+                            output.AppendLine($"    It is referenced in '{referencedFile}'");
+                        }
+                        else
+                        {
+                            ++ countNotFound;
+                            output.AppendLine($"Deleting '{markdownFile.FullName}'.");
+
+                            File.Delete(markdownFile.FullName);
+                        }
                     }
                 }
             }
 
-            output.AppendLine($"\nFound {countNotFound} total .md files that are not referenced in a TOC.\n");
+            var deleted = deleteOrphanedTopics ? "and deleted " : "";
+            output.AppendLine($"\nFound {deleted}{countNotFound} total .md files that are not referenced in a TOC.\n");
```

Comment:
How about we keep a separate count for files that aren't in a TOC but are linked to, and output that info too. Something like "Found 10 .md files that aren't referenced in a TOC. Deleted 8 of these files." or "Found 10 .md files that aren't referenced in a TOC. 2 of these files are linked from other files and weren't deleted."

Reply:
Perfect, I have updated that.